graphql/rules.lua

```diff
@@ -516,6 +516,7 @@ local function isVariableTypesValid(argument, argumentType, context,
     if hasDefault and variableType.__type ~= 'NonNull' then
       variableType = types.nonNull(variableType)
     end
+    if argumentType.kind ~= nil then argumentType = argumentType.kind end
```

Comment:
Hm... Shouldn't (couldn't) it be fixed on top level?

I guess here:
https://github.com/tarantool/graphql/blob/7d2e9c9a1ad580abe92790144b5a78cad01b0ee8/graphql/rules.lua#L544-L545

Currently it looks like a hack to fix a "symptom" but not problem in general.

Reply:
Well, `isVariableTypesValid` is a recursive function if it is called for `'list'` or `'inputObject'`, but if it's called directly for a `'variable'`, there wouldn't be a "top level" like the one you proposed. For example, it happens in this PR test. It will be called directly from `validate.lua` and I'm not sure it's an appropriate place (and not sure that it will work).

Reply:
Well, thanks for explanation